samcli/local/apigw/local_apigw_service.py

```diff
@@ -18,35 +18,85 @@
 
 
 class Route(object):
-
-    def __init__(self, methods, function_name, path, binary_types=None, stage_name=None, stage_variables=None):
+    ANY_HTTP_METHODS = ["GET",
+                        "DELETE",
+                        "PUT",
+                        "POST",
+                        "HEAD",
+                        "OPTIONS",
+                        "PATCH"]
+
+    def __init__(self, function_name, path, method):
         """
         Creates an ApiGatewayRoute
 
         :param list(str) methods: List of HTTP Methods
         :param function_name: Name of the Lambda function this API is connected to
         :param str path: Path off the base url
         """
-        self.methods = methods
+        self.method = method.upper()
         self.function_name = function_name
         self.path = path
-        self.binary_types = binary_types or []
-        self.stage_name = stage_name
-        self.stage_variables = stage_variables
+
+    @staticmethod
+    def normalize_routes(routes):
```

Comment:
Not sure I understand the reason for moving the normalization methods into the Route. Previously the Route was a data holder but now a caller needs to understand which method to call? I think these belong up the call stack and outside of this class (similar to how it was before). But maybe I am missing the need for it here specifically.